### ISSUE TITLE
Implement Weight Decay Optimizers

### ISSUE THREAD

**Issue by seanpmorgan** [tensorflow/addons#24]
Per the [RFC](https://github.com/tensorflow/community/blob/393dfc9e03dd4150a839de6357a26e0259aa97d6/rfcs/20181214-move-to-addons.md), we need to move [Weight Decay Optimizers](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/opt/python/training/weight_decay_optimizers.py) from contrib to addons:

- [ ] DecoupledWeightDecayExtension
- [ ] AdamWOptimizer
- [ ] MomentumWOptimizer

This will involve inheriting from base Keras optimizer, modifying the code to match those APIs, and modifying test cases to run in TF2.

**Comment by seanpmorgan** [comment]
@PhilJd That'd be great... we'd very much appreciate the help!

We're still very early on with setting up contribution guidelines (tests, docs, etc.) so bare with us during review. Also regarding ShampooW, we had not slated the Shampoo optimizer to be moved so that's why the weight decay version is not tracked in this issue.

**Comment by tabshaikh** [comment]
@PhilJd Are you still working on this issue?

**Comment by PhilJd** [comment]
@tabshaikh I've postponed this for a while as I wanted to wait until the CI is up and running. I'll take a look at it tomorrow ;)

**Comment by lgeiger** [comment]
Looks like CI is up and running now. @PhilJd are you still working on this issue?

**Comment by armando-fandango** [comment]
@lgeiger @PhilJd Best way is to create an issue and assign to yourself, and if you are not working on it anymore then un-assign yourself :-) That way we can resolve who is working on what in a distributed way :-)

**Comment by PhilJd** [comment]
Sorry, the last few weeks have been extremely busy and to make matters worse  my machine broke. Now I'm set up again, I've adapted the optimizer, only tests are still missing. I'll be able to finish and open the pull request tomorrow, finally :)
Sorry again!

**Comment by PhilJd** [comment]
I'm still fighting import errors of the built addons (`_image_ops.so` not found), so this might be delayed until tomorrow :(

**Comment by seanpmorgan** [comment]
@PhilJd so that occurs when trying to import tfa (e.g. running a test) using just the repository source code. You'll need to compile the C++ code ahead of time so those .so files can be found. We do this by testing with bazel. Check out the [contributing readme](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md) to see the recommended way to run tests.
